### PR TITLE
Firestore:Fix getAll and get methods implemented with observer

### PR DESCRIPTION
**Pull request by abhinav-qlogic**

### CODE REVIEW

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 1]
```diff
@@ -354,6 +355,15 @@ public T apply(List<T> results) {
     return extractFirst(firestore.getAll(this));
   }
 
+  /** Reads the document referenced by this DocumentReference. */
+  @Nonnull
+  public void get(
+      @Nullable FieldMask fieldMask,
+      ApiStreamObserver responseObserver,
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: Don't we need to specify the type of the `ApiStreamObserver`?

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 2]
```diff
@@ -354,6 +355,15 @@ public T apply(List<T> results) {
     return extractFirst(firestore.getAll(this));
   }
 
+  /** Reads the document referenced by this DocumentReference. */
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: Javadoc should be more descriptive. something like `fetches the document reference in the form of the stream inside `apiStreamObserver`.

File: google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 3]
```diff
@@ -1190,4 +1191,55 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
...
+        FieldMask.of("foo"),
+        new ApiStreamObserver<DocumentSnapshot>() {
+
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            documentSnapshots.add(documentSnapshot);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {}
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: I think confirming this isnt being called would be helpful.

`Assert.fail("No error occurred while streaming");`

File: google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 4]
```diff
@@ -1190,4 +1191,55 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
...
+    ApiStreamObserver responseObserver =
+        new ApiStreamObserver<DocumentSnapshot>() {
+
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            documentSnapshots.add(documentSnapshot);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {}
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: same here

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java (hunk 1 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: type is added

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java [hunk 5]
```diff
@@ -248,6 +248,98 @@ public void onCompleted() {
...
+                documentReference =
+                    new DocumentReference(
+                        FirestoreImpl.this, ResourcePath.create(response.getMissing()));
+                documentSnapshot =
+                    DocumentSnapshot.fromMissing(
+                        FirestoreImpl.this,
+                        documentReference,
+                        Timestamp.fromProto(response.getReadTime()));
+                break;
+              default:
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java:290]: one liner log would be great at default

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java (hunk 2 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: done

File: google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 3 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: done

File: google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 4 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: done

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java (hunk 5 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java:290]: done

File: google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 6]
```diff
@@ -354,6 +355,15 @@ public T apply(List<T> results) {
     return extractFirst(firestore.getAll(this));
   }
 
+  /** fetches the document reference in the form of the stream inside apiStreamObserver */
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: s/fetches/Fetches

Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: done

**Review by rahulKQL — APPROVED** [review]
LGTM

**Pull request closed without merging**